monkey/agent_plugins/exploiters/smb/src/plugin.py

```diff
@@ -0,0 +1,71 @@
+import logging
+from pprint import pformat
+from typing import Any, Dict, Sequence
+from uuid import UUID
+
+# common imports
+from common.event_queue import IAgentEventPublisher
+from common.types import Event
+from common.utils.code_utils import del_key
+
+# dependencies to get rid of or internalize
+from infection_monkey.exploit import IAgentBinaryRepository
+from infection_monkey.i_puppet import ExploiterResultData, TargetHost
+from infection_monkey.propagation_credentials_repository import IPropagationCredentialsRepository
+
+# from .smb_exploit_client import SMBExploitClient
```

Comment:
I don't think we need this commit. We're just adding extra work. It's okay if the UTs fail. That's why we aren't merging to develop.